src/widgets/modals/ByProgramModal.js

```diff
@@ -0,0 +1,406 @@
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { StyleSheet, View } from 'react-native';
+
+import { PageContentModal } from './PageContentModal';
+
+import globalStyles, { DARK_GREY, WARM_GREY, SUSSOL_ORANGE } from '../../globalStyles';
+import { AutocompleteSelector, ToggleBar, PageButton, TextEditor } from '..';
+
+import { SETTINGS_KEYS } from '../../settings';
+import { getAllPrograms, getAllPeriodsForProgram } from '../../utilities/byProgram';
+import SequentialSteps from '../SequentialSteps';
+
+const localization = {
+  program: 'program',
+  supplier: 'supplier',
+  orderType: 'order type',
+  period: 'period',
+  programTitle: 'Select a Program to use',
+  supplierTitle: 'Select a Supplier to use',
+  orderTypeTitle: 'Select a Order Type to use',
+  periodTitle: 'Select a Period to use',
+  nameTitle: 'Give your stocktake a name',
+  programError: 'No programs available',
+  supplierError: 'No suppliers available',
+  orderTypeError: 'No order types available',
+  periodError: 'No periods available',
+};
+
+const getNewState = command => {
```

Comment:
this can just be an object 
```
return { 
   'RESET_ALL': { 
       program: {},
        supplier: {},
        period: {},
        periods: null,
        orderType: {},
        orderTypes: null,
        name: null,
        currentStep:  
    }, ...
}[command] ||  { modalIsOpen: false };
...

```

Reply:
Done - couldn't quite understand your pseudo though c00254b837aeed77c1808616114b36d7cf209fee